qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/diagnostics/QuteDiagnosticsInExpressionWithSwitchWhenSectionTest.java

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.diagnostics;
+
+import static com.redhat.qute.QuteAssert.d;
+import static com.redhat.qute.QuteAssert.testDiagnosticsFor;
+
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test with expressions in case of switch and when sections.
+ *
+ *
+ */
+public class QuteDiagnosticsInExpressionWithSwitchWhenSectionTest {
+
+	@Test
+	public void switchExpressionTypeMismatch() {
+		String template = "{@org.acme.Item item}\r\n" + //
+				"		{#switch item.name}\r\n" + //
+				"		{#case \"Apple\"}\r\n" + //
+				"		{#case 123}\r\n" + //
+				"		{/switch}";
+		testDiagnosticsFor(template, //
+				d(3, 9, 3, 12, QuteErrorCode.UnexpectedMemberType,
+						"Unexpected type `java.lang.Integer` in `item.name`. Expected `java.lang.String`.",
+						DiagnosticSeverity.Error));
+	}
+	
+	@Test
+	public void switchExpressionTypeMismatchObjectPart() {
+		String template = "{@java.lang.String name}\r\n" + //
+				"		{#switch name}\r\n" + //
+				"		{#case \"Apple\"}\r\n" + //
+				"		{#case 123}\r\n" + //
+				"		{/switch}";
+		testDiagnosticsFor(template, //
+				d(3, 9, 3, 12, QuteErrorCode.UnexpectedMemberType,
+						"Unexpected type `java.lang.Integer` in `name`. Expected `java.lang.String`.",
+						DiagnosticSeverity.Error));
+	}
+	
+	@Test
+	public void switchExpressionTypeMismatchSectionSwitch() {
+		String template = "{@java.lang.String name}\r\n" + //
+				"		{#switch name}\r\n" + //
+				"		{#is \"Apple\"}\r\n" + //
+				"		{/switch}";
+		testDiagnosticsFor(template, //
+				d(1, 3, 1, 10, QuteErrorCode.MismatchingSection,
+						"`is` section must be hosted in a `when` section.",
+						DiagnosticSeverity.Error));
+	}
+	
+	@Test
+	public void switchExpressionTypeMismatchSectionWhen() {
+		String template = "{@java.lang.String name}\r\n" + //
+				"		{#when name}\r\n" + //
+				"		{#case \"Apple\"}\r\n" + //
+				"		{/when}";
+		testDiagnosticsFor(template, //
+				d(1, 3, 1, 8, QuteErrorCode.MismatchingSection,
```

Comment:
Are you sure that it is forbidden? I suggest that you try to write a qute template with this usecase and rendez the template and see there are some errors

Reply:
This template is valid. You can mix #case and #is in #switch or #when. Please don't report error in this test.